logstash-core/lib/logstash/output_delegator_strategies/legacy_worker.rb

```diff
@@ -0,0 +1,27 @@
+# Remove this in Logstash 6.0
+module LogStash module OutputDelegatorStrategies class LegacyWorker
+  ::LogStash::OutputDelegatorStrategyRegistry.instance.register(:legacy_worker, self)
+  attr_reader :worker_count, :workers
+  
+  def initialize(logger, klass, metric, xopts={}, plugin_args={})
+    @worker_count = (plugin_args["workers"] || 1).to_i
+    @workers = @worker_count.times.map {|t| klass.new(plugin_args)}
+    @worker_queue = SizedQueue.new(@worker_count)
+    @workers.each {|w| @worker_queue << w}
+  end
+  
+  def register
+    @workers.each(&:register)
+  end
+  
+  def multi_receive(events)
+    worker = @worker_queue.pop
+  ensure
```

Comment:
Did you mean to not do anything with events? To be consistent with A) th previous code and B) the other strategies, consider this:
```ruby
  def multi_receive(events)
    output = @worker_queue.pop
    begin
      output.multi_receive(events)
    ensure
      @worker_queue.push(output)
    end
  end
```